services/QuillLMS/app/workers/student_learning_sequences/fix_replay_scoring_worker.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module StudentLearningSequences
+  class FixReplayScoringWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: SidekiqQueue::MIGRATION
+
+    def perform
+      replayed_diagnostics.each do |row|
+        FindAndSetCompletedPlaythroughWorker.perform_async(row['user_id'], row['classroom_unit_id'], row['activity_id'])
+      end
+    end
+
+    private def replayed_diagnostics
+      RawSqlRunner.execute(
+        <<-SQL
+          SELECT user_id, classroom_unit_id, activity_id
+            FROM activity_sessions
+            WHERE activity_id IN (2537, 2539, 2541, 2550, 2555, 2563, 1663, 1668, 1678, 1161, 1568, 1590, 1664, 2538, 1669, 2540, 1680, 2542, 1774, 2551, 1814, 2557, 1818, 2564)
+            GROUP BY user_id, classroom_unit_id, activity_id
+            HAVING COUNT(*) > 1
```

Comment:
What about adding this condition on the group by:
`HAVING SUM( (completed_at IS NOT NULL)::int ) > 0`
That way we're restricting the job queue to only candidates with at least one completed sessions.